src/decode.rs

```diff
@@ -358,7 +359,7 @@ unsafe fn read_tx_tree(
     };
 }
 
-fn neg_deinterleave(diff: c_int, r#ref: c_int, max: c_int) -> c_int {
+fn neg_deinterleave(diff: u8, r#ref: u8, max: u8) -> u8 {
```

Comment:
We can't change these to unsigned, `max - 1` may be negative.

Reply:
Does #1056 solve it?